src/tests/ftest/util/environment_utils.py

```diff
@@ -192,6 +195,16 @@ def set_defaults(self, logger, servers=None, clients=None, provider=None, insecu
             self.agent_config = os.path.join(self.log_dir, "configs", "daos_agent.yml")
         if self.server_config is None:
             self.server_config = os.path.join(self.log_dir, "configs", "daos_server.yml")
+        if self.env_file is None:
+            self.env_file = os.path.join(self.log_dir, "configs", "daos_test_env.txt")
+
+        if not os.path.exists(self.env_file):
+            # Write the current environment variables to the test environment file
+            os.makedirs(os.path.dirname(self.env_file), exist_ok=True)
+            with open(self.env_file, 'w', encoding='utf-8') as _env_file:
+                for key, value in os.environ.items():
+                    if key not in EXCLUDED_DAOS_TEST_ENV_VARS:
+                        _env_file.write(f"{key}={value}\n")
```

Comment:
FYI that `value` might need to be quoted in cases such as spaces or special characters.
E.g.
```
$ cat ./env
aaa=ccc ddd
$ source ./env 
bash: ddd: command not found
```